caddy/caddymain/run.go

```diff
@@ -40,6 +40,8 @@ func init() {
 	flag.StringVar(&revoke, "revoke", "", "Hostname for which to revoke the certificate")
 	flag.StringVar(&serverType, "type", "http", "Type of server to run")
 	flag.BoolVar(&version, "version", false, "Show version")
+	flag.StringVar(&caddytls.HTTPChallengePort, "external-port-80-is-local-port", "80", "Local port exposed externally as port 80")
+	flag.StringVar(&caddytls.TLSSNIChallengePort, "external-port-443-is-local-port", "443", "Local port exposed externally as port 443")
```

Comment:
By "I don't think these should change," I mean that those are consts which represent the ports defined by the ACME spec. Flags don't change the spec and I don't think we should be modifying those consts either.

> Do you mean that Caddy should never try to open local ports other than 80 or 443 to respond to ACME challenges?

No. It already opens an alternate port for the HTTP challenge when the server is already running. But you're asking to change the _default_ port. But that is hard-coded into the spec.

> I'm specifically interested in responding to ACME challenges on alternative ports because we have a NAT router forwarding i.e. external_ip:80 to internal_ip:8080.

Only ACME HTTP challenges or all HTTP traffic? I guess I'm not sure clear on what problem you're trying to solve, between this PR and the linked issue which talks about the HTTP port for everything.